pkg/workspace/controllers/workspace_controller.go

```diff
@@ -789,3 +807,59 @@ func (c *WorkspaceReconciler) watchNodeClaims() handler.EventHandler {
 			}
 		})
 }
+
+func (r *WorkspaceReconciler) monitorWorkspaces() {
```

Comment:
Suggest moving this entire monitoring code out to another file. Try to use a standard alone method instead of a method of WorkspaceReconciler struct.